AxonDeepSeg/segment.py

```diff
@@ -73,11 +76,38 @@ def generate_default_parameters(type_acquisition, new_path):
 
     return path_model
 
+def get_model_native_resolution_and_patch(path_model):
+    '''
+    Get the native resolution of the model, ie. the resolution that segmented images will get resampled to.
+    Also, get the patch size.
+    :param path_model: model directory
+    :return: model_resolution, patch_size
+    '''
+    resolution_unit_conversion_factor = 1e3 # IVADOMED uses a native mm resolution, whereas ADS uses um
+    
+    model_json = [pos_json for pos_json in os.listdir(path_model) if pos_json.endswith('.json')]
+    model = json.load(open(path_model / model_json[0]))
+
+    model_resolution = [model['transformation']['Resample']['wspace'], model['transformation']['Resample']['hspace']]
+    model_resolution[0] = model_resolution [0] * resolution_unit_conversion_factor
+    model_resolution[1] = model_resolution [1] * resolution_unit_conversion_factor
+    
+    patch_size = [model['default_model']['length_2D'][0], model['default_model']['length_2D'][1]]
+
+    if model_resolution[0] == model_resolution[1]: #isotropic pixels
+        model_resolution = model_resolution[0]
+
+    if patch_size[0] == patch_size[1]: #isotropic pixels
+        patch_size = patch_size[0]
```

Comment:
See comments below, the code throws errors when there is a mix of float (isotropic) and list (anisotropic) values coming from here.